substrate/frame/balances/src/impl_currency.rs

```diff
@@ -535,6 +535,34 @@ where
 	}
 }
 
+/// Validates whether an account can create a reserve without violating
+/// liquidity constraints.
+///
+/// This method performs liquidity checks without modifying the account state.
+fn ensure_can_reserve<T: Config<I>, I: 'static>(
+	who: &T::AccountId,
+	value: T::Balance,
+	check_existential_deposit: bool,
+) -> DispatchResult {
+	let AccountData { free, reserved, frozen, .. } = Pallet::<T, I>::account(who);
+
+	// Early validation: Check sufficient free balance
+	let new_free_balance = free.checked_sub(&value).ok_or(Error::<T, I>::InsufficientBalance)?;
+
+	// The reserve is allowed as long as the total balance (free + reserved)
+	// remains above the frozen balance
+	let total_balance = free.saturating_add(reserved);
+	ensure!(total_balance >= frozen, Error::<T, I>::LiquidityRestrictions);
+
+	// Conditionally validate existential deposit preservation
+	if check_existential_deposit {
```

Comment:
A reserve can create a consumer ref which makes the ED unnecessary. Not sure if we ever need to check it.

Reply:
> I am slightly in favour of fixing the proxy pallet, but if that will not make it in time then we can still use this fix here...
> With the chance of introducing some subtle bug that we then have to fix again later (which could be an OK trade-off in this case).

IMO this is a more than OK trade-off in this case. This bug affects all pallets that still use Currency. For example, you can't place the DD from an account that has frozen > free for the same reason. Obviously the proper solution is to migrate the pallets to use Fungible, but since this will take some time I think this "dirty" fix is the way to go, even if there are subtle bugs introduced. Unless there are reasons to believe it's going to be worse than now. 😅

Reply:
> I think its safer to always respect the ED. on reserve too.

In fungible, free_balance must be `>=` to ED as long as there is no other provider. But I don't know if `Currency` currently respect this property.
I think we can safely always check for E.D. when reserving from now on. But this might break some tests indeed.

Reply:
Having the ED account preservation checked in the `reserve` call would be more intrusive. But if we end up choosing that approach, the helper logic can be moved to `can_reserve` and `reserve` then can call `call_reserve`.